### ISSUE TITLE
Onboarding repository `mapbox/vector-tile` to `maplibre/mvt-cpp`

### ISSUE THREAD

**Issue by nathreed** [maplibre/maplibre#366]
https://github.com/louwers/mvt-cpp

There is no MapLibre GitHub URL yet. This is a proposal for MapLibre to fork https://github.com/mapbox/vector-tile.

## Motivation

https://github.com/mapbox/vector-tile is a dependency of (at least) MapLibre Native, used for decoding MVT. This is a proposal for MapLibre to fork it and do minimal maintenance going forward, since it appears to be relatively unmaintained/stagnant on the upstream side, and there are some outstanding issues that impact MapLibre Native. 

Specifically, the upstream version throws exceptions when processing certain tiles, in apparent violation of the MVT spec (see https://github.com/mapbox/vector-tile/issues/55 and https://github.com/maplibre/maplibre-native/issues/795). This issue has been outstanding for more than five years, and two PRs to fix were provided by the original reporter (@xabbu42). However, it seems like momentum to merge them was lost after discussions about the performance characteristics of the two options and debate over whether this should be fixed in PostGIS (it wasn't, and there are reports of tiles from other sources e.g. GDAL having this issue as well).

Onboarding this repository would be good for the MapLibre project because it would allow the project to have control over an important dependency for MapLibre Native and apply fixes for longstanding issues. The library appears quite simple overall, and is rather self-contained and mature in terms of functionality, so it hopefully would not be a large ongoing burden to maintain, and no new functionality would need to be added -- only bug fixes. It's important for MapLibre Native to work with standard-conforming tiles created from a diversity of sources, and forking `vector-tile` to apply these fixes would improve the current state with regard to that goal. 

## Acceptance
- [x] Any two board members must agree to accept a new repository.
  **Approved by:** @HarelM, @birkskyum 

## Licensing
- [ ] The repo license is BSD-3 or MIT.
  *Repos may allow dual-licensing under other open source licenses, e.g. MIT OR Apache.*

It's ISC:

> The ISC license is functionally equivalent to the BSD 2-Clause and MIT licenses, removing some language that is no longer necessary.

- [x] The repo contains `Copyright (c) <year> MapLibre contributors` in license file(s) 
- [ ] and in the readme.

## Special files
- [x] `/README.md`
  *Description, link to the main maplibre.org page, name of the OSM-US Slack channel for discussions and an [invite link](https://slack.openstreetmap.us), etc*
- [x] `/LICENSE`
   *Dual-licensed repos may have additional files like `LICENSE-MIT` and `LICENSE-APACHE`*
- [x] `/SECURITY_POLICY.txt`
    Add the text:
    ```
    For an up-to-date policy refer to
    https://github.com/maplibre/maplibre/blob/main/SECURITY_POLICY.txt
    ```
- [x] `/CONTRIBUTING.md`
- [ ] The repo has Pull Request and Issue Templates in `/.github` dir.
- [x] The repo has `/.github/FUNDING.yml` file copied from [maplibre-gl-js/funding](https://github.com/maplibre/maplibre-gl-js/blob/main/.github/FUNDING.yml)
- [x] `/CODE_OF_CONDUCT.md`
  *This file should only link to our [primary code of conduct](https://github.com/maplibre/maplibre/blob/main/CODE_OF_CONDUCT.md). Use this markup for consistency:*
 `# Contributor Covenant`
 `[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](https://github.com/maplibre/maplibre/blob/main/CODE_OF_CONDUCT.md)`

#### General page
- [x] **[Features]** Disable unused features like wiki.
- [x] **[Features]** Enable `Sponsorships` checkbox (see also FUNDING.yaml above).
- [x] **[Features]** Enable `Preserve this repository`.
- [x] **[Pull Requests]** Community is encouraged to use `squash merge`. Disable other merge types if possible.
- [x] **[Pull Requests]** Enable `Automatically delete head branches`.

#### Access
- [x] The repo has at least one admin who is ideally not part of the Governing Board: @louwers 

#### Branches
- [x] The primary branch is named `main`.
- [x] Set up branch rules to require CI pass and an approval before merge.
  *For smaller projects it might be OK to ignore this rule.*

## Miscelaneous
- [x] The repo must not have any personal branches.
  *All work should be done on forks and submitted via PRs, including by the admins.*
- [x] Repo has a proper GitHub description and an optional web site
  *Use the gear icon in the upper right corner of the repo page.*
- [x] CI automatically runs on all pull requests before merging using GitHub actions
- [ ] ~Grant admin rights to the board members and automation accounts for packages~ N/A

## Community
- [ ] The new repo has been announced in the `#maplibre` OSMUS slack channel.
- [ ] The new repo has been announced in the next monthly meeting of the Technical Steering Committee.
- [ ] The new repo has been announced in the `@maplibre` twitter.

**Comment by louwers** [comment]
We need to fork it before we can onboard it.

**Comment by louwers** [comment]
Do we want to get rid of the npm package? I don't think any MapLibre projects use it?

https://www.npmjs.com/package/@mapbox/vector-tile

**Comment by louwers** [comment]
To avoid confusion with the new MapLibre Tile Specification, I suggest renaming the library to `mvt-cpp`.

**Comment by louwers** [comment]
It looks like we can go ahead and transfer this.

I will make a PR to rename some directories and update MapLibre Native to use the fork after the transfer.

**Comment by birkskyum** [comment]
OK, @louwers I imported it to https://github.com/maplibre/mvt-cpp and made you admin of it.

**Comment by louwers** [comment]
@nathreed Could you open pull request with the fix you were looking to get merged? Thanks!
